sys/riscv/riscv/freebsd64_machdep.c

```diff
@@ -246,7 +246,7 @@ freebsd64_sendsig(sig_t catcher, ksiginfo_t *ksi, sigset_t *mask)
 	onstack = sigonstack(tf->tf_sp);
 
 	CTR4(KTR_SIG, "sendsig: td=%p (%s) catcher=%p sig=%d", td, p->p_comm,
-	    catcher, sig);
+	    (__cheri_fromcap void *)catcher, sig);
```

Comment:
Probably this needs to be __cheri_addr vaddr_t or the like.  KTR internally stores an array of longs and relies on the fact that all arches promote non-long types to long for var_args.  Not sure what the pure capability kernel does (@qwattash probably knows), though I suspect just bumping it to uintcap_t won't DTRT because we don't do var-args that way but only promote them to longs instead of uintcap_t.  (Though our lives would probably be simpler overall if we just rounded up all var-args to an array of uintcap_t which I've mentioned to @jrtc27 even though it's a bit more wasteful of stack space)

Reply:
I have not been including KTR in recent kernels but I believe we are currently using an array of uintptr_t entries for the arguments. The actual CTRx macro does not use a variadic function so we should not see that issue there, however I think that the issue @jhb is describing happens down in ddb when trying to print out KTR arguments, and possibly in KTR_ALQ. Anyway I don't think we want to be passing capabilities down to KTR, so I think we should make those vaddr_t.

Reply:
Is there an ETA on `ptraddr_t` being available or should I just use `vaddr_t` and we can change them all in a follow-up commit?

Reply:
I think for now keep using vaddr_t and I'll do a big search+replace change later. Possibly next week, but could also take longer.